### PR TITLE
clang/tidy: Fixes (`/test`)

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
this is currently breaking windows:

```console
# Execution platform: @envoy_build_tools//toolchains:rbe_windows_msvc_cl_platform
original_src_test.obj : error LNK2019: unresolved external symbol "public: __cdecl Envoy::MockBufferBase<class Envoy::Buffer::OwnedImpl>::MockBufferBase<class Envoy::Buffer::OwnedImpl>(void)" (??0?$MockBufferBase@VOwnedImpl@Buffer@Envoy@@@Envoy@@QEAA@XZ) referenced in function "public: __cdecl Envoy::MockBuffer::MockBuffer(void)" (??0MockBuffer@Envoy@@QEAA@XZ)
bazel-out\x64_windows-opt\bin\test\extensions\filters\http\original_src\original_src_test.exe : fatal error LNK1120: 1 unresolved externals
INFO: Elapsed time: 236.233s, Critical Path: 151.14s
INFO: 5785 processes: 2931 remote cache hit, 2820 internal, 34 remote.
//test/extensions/filters/http/original_src:original_src_test   FAILED TO BUILD
```

**Comment by wbpcode** [comment]
Although all these are test only changes, I am still scared, a bit.

Will take a look tomorrow.

**Review by wbpcode — APPROVED** [review]
LGTM with only two nit comments (you can decide update or just ignore them). Thanks for you so great contribution. I know this need lots of time and energy. Thanks.

File: test/extensions/quic/connection_id_generator/envoy_deterministic_connection_id_generator_test.cc [hunk 1]
```diff
@@ -25,7 +24,7 @@ class EnvoyDeterministicConnectionIdGeneratorTest : public QuicTest {
         generator_(EnvoyDeterministicConnectionIdGenerator(connection_id_length_)) {}
 
 protected:
-  int connection_id_length_;
+  int connection_id_length_ = 0;
```
[the remark this answers is not in the document]
Comment (phlax) [reply test/extensions/quic/connection_id_generator/envoy_deterministic_connection_id_generator_test.cc:27]: actually - lets land - i have another PR behind with all the leftovers, and less clear cases so i ill revert these and retest in that

**Pull request merged**